globus_automate_client/cli/flows.py

```diff
@@ -57,6 +60,30 @@ class ActionStatus(str, Enum):
     inactive = "INACTIVE"
 
 
+class FlowInputFormat(str, Enum):
+    json = "json"
+    yaml = "yaml"
+
+
+def _process_definition(definition: str, input_format: str) -> Mapping[str, Any]:
+    """
+    Turn input strings into dicts per input format type (FlowInputFormat)
+    """
+    flow_dict = None
+    if input_format == FlowInputFormat.json:
```

Comment:
There was a whole discussion on the philosophy of `Enum` in #devchat back in November, from which, to be honest, I'm not certain what the main conclusions were. That said, I would argue for this sort of usage:
```suggestion
    if FlowInputFormat(input_format) is FlowInputFormat.json:
```
which has the benefits of
- Generating an informative `ValueError` if `input_format` is somehow not in the enum
- Making a comparison that's within the universe of your enum type rather than comparing string values, which is a "value" comparison rather than a "type" comparison, to be really picky about it

To that effect, I might even argue for changing the signature to
```python
_process_definition(definition: str, input_format: FlowInputFormat)
```
and here using
```suggestion
    if input_format is FlowInputFormat.json:
```
while calling `_process_definition(definition, FlowInputFormat(some_string_value))` elsewhere.

Also roping in @urimandujano and @jimPruyne for this discussion, sorry to blow up this one bit of code @jakeglobus but felt it's worth us having this discussion as it relates our codebases and this is a decent example.

Reply:
I remember that :) I'm not sure it's as valuable to generate an informative `ValueError` here since typer will try to do that itself when it parses the CLI argument. But the verdict's still out on other places we use Enums. I do like the `is` optional for readability though.